packages/ai/src/LLMStep.ts

```diff
@@ -97,7 +97,7 @@ export class LLMStepInstance<const Shape extends StepShape = StepShape> {
   ) {
     this.startTime = Date.now();
     this.id = crypto.randomUUID();
-    this.logger = new Logger();
+    this.logger = new Logger(this.workflow.id, this.workflow.getStepCount());
```

Comment:
I'll have to partially revert this for my upcoming PR - in it, `LLMStepInstance` is not parameterized with `workflow`, to avoid circular references, I pass the workflow in `step.run()` as a parameter instead.

But I see a way how to do that and keep the features you cared about here.